src/server/enterprise/cmds/cmds.go

```diff
@@ -149,8 +156,10 @@ func RegisterCmd() *cobra.Command {
 		}),
 	}
 	register.PersistentFlags().StringVar(&id, "id", "", "the id for this cluster")
-	register.PersistentFlags().StringVar(&pachdAddr, "pachd-address", "", "the address for the enterprise server to reach this pachd")
+	register.PersistentFlags().StringVar(&pachdAddr, "pachd-address", "", "the address for a user to reach this pachd")
+	register.PersistentFlags().StringVar(&pachdEntAddr, "pachd-enterprise-address", "", "the address for the enterprise server to reach this pachd")
 	register.PersistentFlags().StringVar(&enterpriseAddr, "enterprise-server-address", "", "the address for the pachd to reach the enterprise server")
+	register.PersistentFlags().StringVar(&clusterId, "cluster-deployment-id", "", "the deployment id of the cluster being registered")
```

Comment:
The admin running this command has access to the cluster being registered (from `client.NewOnUserMachine`). If they don't specify a cluster ID we should be able to get it from https://github.com/pachyderm/pachyderm/blob/master/src/admin/admin.proto#L15